cfme/utils/log_validator.py

```diff
@@ -77,12 +78,22 @@ def _check_match_logs(self, line):
         for pattern in self.matched_patterns:
             if re.search(pattern, line):
                 logger.info('Expected pattern {} was matched on line {}'.format(pattern, line))
-                self.matches[pattern] = True
+                self.matches[pattern] = self.matches[pattern] + 1
 
-    def _verify_match_logs(self):
-        for pattern in self.matched_patterns:
-            if pattern not in self.matches:
-                pytest.fail('Expected pattern {} did not match'.format(pattern))
+    def validate_logs(self):
+        """Validate log pattern"""
+        self._validation()
+        for pattern, count in self.matches.items():
+            if count == 0:
+                pytest.fail(
+                    'Expected pattern {} did not match; match count {}'.format(pattern, count)
+                )
+
+    @property
+    def patterns_match_count(self):
```

Comment:
I think the naming is a bit confusing here, it does not read clearly at all for me, but maybe its just me.

Consider changing the naming pattern/scheme here to be more clear, maybe using `number_of` or `count_of` in some way?

```
@property
def count_of_pattern_matches

def wait_for_pattern_count_match()
```

I think the docblocks need more examples, because its not obvious what is returned in the dictionary.

Reply:
Considering my comment below about an alternative design, I think this is a wrapper around something that should be a property anyway.

I think instead of this extra property, `LogValidator.matches` should be a property, which calls `self._validation` and then returns `self._matches`, which would be updated in `_check_match_logs` (instead of `self.matches` like now).

Reply:
@mshriver now
`LogValidator.matches` is property... which used while validation. and `_matches` is for count collection.